src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -251,8 +251,8 @@ internal sealed class SourcePropertySymbol : PropertySymbol, IAttributeTargetSym
 
                         // Although we only do this in error scenarios, it is undesirable to mutate the symbol by setting its type twice.
                         // Tracked by https://github.com/dotnet/roslyn/issues/35381
-                        _lazyType.InterlockedDangerousReset();
-                        _lazyType.InterlockedInitialize(type);
+                        Interlocked.Exchange(ref _lazyType, null);
+                        Interlocked.CompareExchange(ref _lazyType, new TypeWithAnnotations.Boxed(type), null);
```

Comment:
>Interlocked.CompareExchange(ref _lazyType, new TypeWithAnnotations.Boxed(type), null); [](start = 24, length = 86)

It looks like Interlocked operation is not necessary here, we are in a constructor and other threads shouldn't be able to access the instance before it is fully initialized.